src/metabase/server/middleware/security.clj

```diff
@@ -124,11 +125,77 @@
           "Content-Security-Policy"
           #(format "%s frame-ancestors %s;" % (if allow-iframes? "*" (or (embedding-app-origin) "'none'")))))
 
+(defn parse-url
+  "Returns an object with protocol, domain and port for the given url"
+  [url]
+  (if (= url "*")
+    {:protocol nil :domain "*" :port "*"}
+    (let [pattern #"^(?:(https?)://)?([^:/]+)(?::(\d+|\*))?$"
+          matches (re-matches pattern url)]
+      (if matches
+        (let [[_ protocol domain port] matches]
+          {:protocol protocol
+           :domain domain
+           :port port})
+        (throw (IllegalArgumentException. (str "Invalid URL '" url "'")))))))
```

Comment:
I was thinking, maybe we could just skip throwing the exception since all we're doing is logging it? Maybe something like:

```
(if-not matches
 (do (log/errorf "Invalid URL: %s" url) nil)
 (let ...)
```

Then later we can just do `(keep parse-url urls)` without needing to worry about having a `try/catch`. WDYT?

Reply:
Makes sense, done in 0435edec07d4e2e08896f91e011ae1ceded5832d